test/fakeintake/client/flare/parse_flare.go

```diff
@@ -0,0 +1,198 @@
+	Content-Disposition: form-data; name="agent_version"
+	Content-Type: text/plain
+
+	7.45.0
+	--0cf50bf933f0ddecdd8cbfada84c4d6fa4cb226fd5893493f51fd01dfea5
+	Content-Disposition: form-data; name="hostname"
+	Content-Type: text/plain
+
+	hostname123
+	--0cf50bf933f0ddecdd8cbfada84c4d6fa4cb226fd5893493f51fd01dfea5--
+*/
+// In this example the pattern '0cf50bf933f0ddecdd8cbfada84c4d6fa4cb226fd5893493f51fd01dfea5' is the second argument `boundary` and
+// each content between two boundaries is fetched with `mimeReader.NextPart()` in the function below.
+// The boundary is provided by the initial flare request from the Agent in the `Content-Type` header,
+// 	for example: Content-Type: multipart/form-data; boundary=0cf50bf933f0ddecdd8cbfada84c4d6fa4cb226fd5893493f51fd01dfea5
+// and the `Content-Type` value is obtained from the Fake Intake via the "encoding" key
+//
+// The flare zip content is as well parsed and then slightly transformed into a mapping between filenames and File to make querying
+// simpler (e.g. verify that `agent.log` exists and has the right permissions)
+//
+// The resulting output is a Flare struct that provides an API to verify assertions on the flare.
+
+// decodeRawFlare parses the flare payload sent by the Fake Intake into a manageable Flare struct
+// For that it parses the multipart data from the flare request and then parses the flare zip raw content.
+func ParseRawFlare(flarePayload api.Payload) (Flare, error) {
+	// flarePayload.Encoding contains the value of Content-Type header from the flare request
+	boundary := parseBoundaryFromContentTypeHeader(flarePayload.Encoding)
+	parsedFlareData, err := parseFlareMultipartData(string(flarePayload.Data), boundary)
+	if err != nil {
+		return Flare{}, err
+	}
+
+	// flare_file is the only part that needs a special parsing as it's the flare zip content.
+	suffixToTrim := string(parsedFlareData["hostname"]) + string(os.PathSeparator)
+	zipFiles, err := parseRawZIP(parsedFlareData["flare_file"], suffixToTrim)
+	if err != nil {
+		return Flare{}, err
+	}
+
+	return Flare{
+		email:        string(parsedFlareData["email"]),
+		zipFiles:     zipFiles,
+		agentVersion: string(parsedFlareData["agent_version"]),
+		hostname:     string(parsedFlareData["hostname"]),
+	}, nil
+}
+
+// parseBoundaryFromContentTypeHeader parses the value of the Content-Type header sent by the Agent when sending the flare
+// and returns the boundary linked to the multipart data.
+//
+// Example:
+//
+//	Input: "multipart/form-data; boundary=0cf50bf933f0ddecdd8cbfada84c4d6fa4cb226fd5893493f51fd01dfea5"
+//	Output: "0cf50bf933f0ddecdd8cbfada84c4d6fa4cb226fd5893493f51fd01dfea5"
+func parseBoundaryFromContentTypeHeader(contentTypeHeader string) string {
+	var encoding string
+
+	mediaType, params, err := mime.ParseMediaType(contentTypeHeader)
+	if err != nil || !strings.HasPrefix(mediaType, "multipart/") {
+		log.Printf("Could not parse Content-Type header from flare request")
+		encoding = ""
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 warning:** ‏I might be wrong but you likely don't want to return an empty string in case of error but rather return an error or fail the test

Reply:
Returning an error now 👍 411fe3bf4c360d94335f78878775862550d3686f